liquibase-core/src/main/java/liquibase/command/AbstractCommandStep.java

```diff
@@ -1,15 +1,160 @@
 package liquibase.command;
 
+import liquibase.Contexts;
+import liquibase.GlobalConfiguration;
+import liquibase.LabelExpression;
+import liquibase.Scope;
+import liquibase.changelog.ChangeLogHistoryService;
+import liquibase.changelog.ChangeLogHistoryServiceFactory;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.database.Database;
 import liquibase.exception.CommandValidationException;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.MissingRequiredArgumentException;
+import liquibase.integration.commandline.CommandLineUtils;
+import liquibase.integration.commandline.LiquibaseCommandLineConfiguration;
+import liquibase.lockservice.LockServiceFactory;
+import liquibase.resource.ResourceAccessor;
 import liquibase.util.StringUtil;
 
 import java.util.Arrays;
+import java.util.ResourceBundle;
+
+import static java.util.ResourceBundle.getBundle;
 
 /**
  * Convenience base class for {@link CommandStep} implementations.
  */
 public abstract class AbstractCommandStep implements CommandStep {
 
+    private static final ResourceBundle coreBundle = getBundle("liquibase/i18n/liquibase-core");
+
+    private Database database;
+
+    protected Database getDatabase() {
+        return database;
+    }
+
+    private void setDatabase(Database database) {
+        this.database = database;
+    }
+
+    /**
+     * Try to retrieve and set the database object from the command scope, otherwise creates a new one .
+     *
+     * @param commandScope current command scope
+     * @param databaseCommandArgument the CommandArgumentDefinition identifying the expected database object
+     * @throws DatabaseException Thrown when there is a connection error
+     */
+    protected void setOrCreateDatabase(CommandScope commandScope, CommandArgumentDefinition<Database> databaseCommandArgument) throws DatabaseException, CommandValidationException {
```

Comment:
I'm unsure on this method. In general, I would have asked questions about "not all commands have a database", so maybe this should be on a new abstract class like AbstractDatabaseCommandStep" and also how "there are some commands with multiple databases, like diff that has both database and referenceDatabase so maybe make this a function that returns a database given the argument names and returns the Database object so it can be used for both" 

But, since this is hopefully just a temporary thing until we switch to a shared ConnectToDatabaseCommandStep and then this goes away, it probably doesn't really matter. But still mark it as deprecated so external people know not to use it since it's going away? 

Or, we maybe can't even take this partial step of having this function at all? (see overall comment)